zerver/lib/actions.py

```diff
@@ -2776,6 +2780,24 @@ def user_info(um):
     send_event(event, list(map(user_info, ums)))
     return len(changed_messages)
 
+# Calculate the average weekly traffic of a stream
+def get_stream_message_count(stream_id, date_created):
+    # type: (int, datetime.datetime) -> int
+    count = 0
+    stat = COUNT_STATS['messages_in_stream:is_bot:day']
+    d = timezone.now() - datetime.timedelta(days=28)
+    queryset = StreamCount.objects.filter(stream_id=stream_id, property=stat.property,
+                                          end_time__gt=d).values_list('value')
+    for val in queryset:
+        count += val[0]
+    num_days = (timezone.now().date() - date_created.date()).days
+    # Calculates the average over 4 weeks
+    # If duration less than 4 weeks, adjust accordingly
+    if num_days:
+        count = int(old_div(count, 4)) if num_days >= 28 else int(old_div(count, (old_div(num_days, 7))))
```

Comment:
Don't use `old_div`, use `a // b` (the lint error is confused about this)

Also, I get divide-by-0 errors trying this out in a development environment with no new messages.  You should fix that and add tests for the case of no data.